src/notifications/common.ts

```diff
@@ -35,3 +45,66 @@ export enum NotificationPreferenceType {
 export enum NotificationPreferenceStatus {
   Muted = 'muted',
 }
+
+export const notificationPreferenceMap: Partial<
+  Record<NotificationType, NotificationPreferenceType>
+> = {
+  [NotificationType.ArticleNewComment]: NotificationPreferenceType.Post,
+  [NotificationType.CommentReply]: NotificationPreferenceType.Comment,
+  [NotificationType.SquadPostAdded]: NotificationPreferenceType.Source,
+  [NotificationType.SquadMemberJoined]: NotificationPreferenceType.Source,
+};
+
+type NotificationPreferenceUnion = NotificationPreferenceComment &
+  NotificationPreferencePost &
+  NotificationPreferenceSource;
+
+export const saveNotificationPreference = async (
+  con: DataSource | EntityManager,
+  userId: string,
+  referenceId: string,
+  notificationType: NotificationType,
+  status: NotificationPreferenceStatus,
+) => {
+  const type = notificationPreferenceMap[notificationType];
+
+  if (!type) {
+    throw new ValidationError('Notification type not supported');
+  }
+
+  const params: Partial<NotificationPreferenceUnion> = {
+    type,
+    userId,
+    status,
+    notificationType,
+    uniqueKey: referenceId,
+  };
+
+  switch (type) {
```

Comment:
Why don't we use a map here, I think they look slightly cleaner and easier to add to 😂 (but maybe i'm lazy)

Reply:
Not sure how we can do that as we are trying to set some values. Can you give an example?

Reply:
Very quick and dirty, but something like this, let me know what you think.

```js
const type = 'a';
const params = {};
const referenceId = '123';

const typeToParams = {
  'a': 'commentId'
};

params[typeToParams[type]] = referenceId;

console.log(params);
```

Reply:
Let's see how we can turn that to typescript.